buildSrc/src/main/java/com/osacky/flank/gradle/FlankExecutionTask.kt

```diff
@@ -1,19 +1,72 @@
 package com.osacky.flank.gradle
 
+import org.gradle.api.DefaultTask
 import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFile
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import org.gradle.process.ExecOperations
 import javax.inject.Inject
 
-open class FlankExecutionTask @Inject constructor(projectLayout: ProjectLayout, private val config: FladleConfig) : FlankJavaExec(projectLayout) {
+open class FlankExecutionTask @Inject constructor(
+  objectFactory: ObjectFactory,
+  private val projectLayout: ProjectLayout,
+  private val execOperations: ExecOperations
+) : DefaultTask() {
+  @get:Input
+  val dumpShards: Property<Boolean> = objectFactory.property<Boolean>().convention(false)
+
+  @get:InputFile
+  val serviceAccountCredentials: Property<RegularFile> = objectFactory.fileProperty()
+
+  @Suppress("unused")
+  @get:InputFile
+  @PathSensitive(PathSensitivity.RELATIVE)
+  val yamlFile: RegularFileProperty = objectFactory.fileProperty()
+
+  @get:InputFiles
+  @Classpath
+  val fladleConfig = objectFactory.fileCollection()
 
   init {
-    doFirst {
-      checkFilesExist(config)
+    group = JavaExec.TASK_GROUP
+    description = "Runs instrumentation tests using flank on firebase test lab."
+
+    outputs.upToDateWhen {
+      false
```

Comment:
this isn't really require yet because we don't declare any output for this task, it's more a "security" thing to make sure we don't mess it later adding an output before making sure it's absolutely safe, and op-in